core/services/fluxmonitor/flux_monitor_test.go

```diff
@@ -186,291 +163,387 @@ func TestConcreteFluxMonitor_StopWithoutStart(t *testing.T) {
+				data, err := models.ParseJSON([]byte(fmt.Sprintf(`{
+					"result": "%d",
+					"address": "%s",
+					"functionSelector": "0x%x",
+					"dataPrefix": "0x000000000000000000000000000000000000000000000000000000000000000%d"
+				}`, test.polledAnswer, initr.InitiatorParams.Address.Hex(), updateAnswerSelector, reportableRoundID)))
+				require.NoError(t, err)
+
+				rm.On("Create", job.ID, &initr, mock.Anything, mock.MatchedBy(func(runRequest *models.RunRequest) bool {
+					return reflect.DeepEqual(runRequest.RequestParams.Result.Value(), data.Result.Value())
+				})).Return(&run, nil)
+
+				fluxAggregator.On("GetMethodID", "updateAnswer").Return(updateAnswerSelector, nil)
+			}
 
-	err = deviationChecker.Start(context.Background())
-	require.NoError(t, err)
-	require.Len(t, jobRunCreated, 0, "no Job Runs created")
+			checker, err := fluxmonitor.NewPollingDeviationChecker(store, fluxAggregator, initr, rm, fetcher, time.Second)
+			require.NoError(t, err)
 
-	require.Eventually(t, func() bool { return len(jobRunCreated) >= 1 }, time.Second, time.Millisecond, "idleThreshold triggers Job Run")
-	require.Eventually(t, func() bool { return len(jobRunCreated) >= 5 }, time.Second, time.Millisecond, "idleThreshold triggers succeeding Job Runs")
+			if test.connected {
+				checker.OnConnect()
+			}
 
-	deviationChecker.Stop()
+			checker.ExportedPollIfEligible(test.threshold)
 
-	assert.Equal(t, decimal.NewFromInt(100).String(), deviationChecker.ExportedCurrentPrice().String())
+			fluxAggregator.AssertExpectations(t)
+			fetcher.AssertExpectations(t)
+			rm.AssertExpectations(t)
+		})
+	}
 }
 
-func TestPollingDeviationChecker_StartError(t *testing.T) {
+func TestPollingDeviationChecker_TriggerIdleTimeThreshold(t *testing.T) {
 	store, cleanup := cltest.NewStore(t)
 	defer cleanup()
 
-	rm := new(mocks.RunManager)
-	job := cltest.NewJobWithFluxMonitorInitiator()
-	initr := job.Initiators[0]
-	initr.ID = 1
-
-	fluxAggregator := new(mocks.FluxAggregator)
-	fluxAggregator.On("LatestAnswer", initr.InitiatorParams.Precision).
-		Return(decimal.NewFromInt(0), errors.New("deliberate test error"))
+	tests := []struct {
+		name             string
+		idleThreshold    time.Duration
+		expectedToSubmit bool
+	}{
+		{"no idleThreshold", 0, false},
+		{"idleThreshold > 0", 10 * time.Millisecond, true},
+	}
 
-	checker, err := fluxmonitor.NewPollingDeviationChecker(store, fluxAggregator, initr, rm, nil, time.Second)
-	require.NoError(t, err)
-	require.Error(t, checker.Start(context.Background()))
-}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			runManager := new(mocks.RunManager)
+			fluxAggregator := new(mocks.FluxAggregator)
+			fetcher := new(mocks.Fetcher)
 
-func TestPollingDeviationChecker_StartStop(t *testing.T) {
-	store, cleanup := cltest.NewStore(t)
-	defer cleanup()
+			job := cltest.NewJobWithFluxMonitorInitiator()
+			initr := job.Initiators[0]
+			initr.ID = 1
+			initr.PollingInterval = models.Duration(math.MaxInt64)
```

Comment:
What happens if someone ever passes 0 in as a PollingInterval? Probably want to validate this.

Reply:
Handled in `pollingDeviationCheckerFactory#New`